ports/servoshell/prefs.rs

```diff
@@ -181,393 +183,233 @@ pub fn read_prefs_map(txt: &str) -> HashMap<String, PrefValue> {
-                    Err(_) => Some(OutputOptions::FileName(argument)),
-                },
-            },
-            None => Some(OutputOptions::Stdout(5.0)),
-        }
-    } else {
-        // if the p option doesn't exist:
-        None
-    };
+    /// Path to an output image. The format of the image is determined
+    /// by the extension. Supports all formats that rust-image does.
+    #[bpaf(short('o'), long)]
+    output_image: Option<String>,
 
-    if let Some(ref time_profiler_trace_path) = opt_match.opt_str("profiler-trace-path") {
-        let mut path = PathBuf::from(time_profiler_trace_path);
-        path.pop();
-        if let Err(why) = fs::create_dir_all(&path) {
-            error!(
-                "Couldn't create/open {:?}: {:?}",
-                Path::new(time_profiler_trace_path).to_string_lossy(),
-                why
-            );
-        }
-    }
+    /// Time profiler flag and either a TSV output filename OR and interval
+    /// for output to Stdout
+    #[bpaf(short('p'), long, argument::<String>("10 OR time.tsv"), parse(parse_profiling), optional)]
+    profile: Option<OutputOptions>,
 
-    let layout_threads: Option<usize> = opt_match.opt_str("y").map(|layout_threads_str| {
-        layout_threads_str
-            .parse()
-            .unwrap_or_else(|err| args_fail(&format!("Error parsing option: -y ({})", err)))
-    });
-
-    let nonincremental_layout = opt_match.opt_present("i");
-
-    let random_pipeline_closure_probability = opt_match
-        .opt_str("random-pipeline-closure-probability")
-        .map(|prob| {
-            prob.parse().unwrap_or_else(|err| {
-                args_fail(&format!(
-                    "Error parsing option: --random-pipeline-closure-probability ({})",
-                    err
-                ))
-            })
-        });
+    /// Path to dump a self-contained HTML timeline of profiler traces
+    #[bpaf(long("profiler-trace-path"))]
+    time_profiler_trace_path: Option<String>,
 
-    let random_pipeline_closure_seed =
-        opt_match
-            .opt_str("random-pipeline-closure-seed")
-            .map(|seed| {
-                seed.parse().unwrap_or_else(|err| {
-                    args_fail(&format!(
-                        "Error parsing option: --random-pipeline-closure-seed ({})",
-                        err
-                    ))
-                })
-            });
-
-    if opt_match.opt_present("devtools") {
-        let port = opt_match
-            .opt_str("devtools")
-            .map(|port| {
-                port.parse().unwrap_or_else(|err| {
-                    args_fail(&format!("Error parsing option: --devtools ({})", err))
-                })
-            })
-            .unwrap_or(preferences.devtools_server_port);
-        preferences.devtools_server_enabled = true;
-        preferences.devtools_server_port = port;
-    }
+    /// A preference to set
+    #[bpaf(argument("dom_bluetooth_enabled"),fallback(vec![]))]
```

Comment:
You don't need `fallback(vec![])` here I think. If you want to be explicit - you can say  

```rust
#[bpaf(argument("dom_bluetooth_enabled"), many)]
pref: Vec<String>,
```

Same with `prefs_files` below.